projects/aca-content/src/lib/store/effects/upload.effects.ts

```diff
@@ -49,15 +49,14 @@ export class UploadEffects {
   private readonly fileVersionInput: HTMLInputElement;
   private readonly uploadMenuButtonSelector = 'app-toolbar-menu button[id="app.toolbar.upload"]';
 
-  constructor(
-    private store: Store<AppStore>,
-    private actions$: Actions,
-    private ngZone: NgZone,
-    private uploadService: UploadService,
-    rendererFactory: RendererFactory2,
-    private contentService: ContentManagementService
-  ) {
-    const renderer = rendererFactory.createRenderer(null, null);
+  private store = inject(Store<AppStore>);
+  private actions$ = inject(Actions);
+  private ngZone = inject(NgZone);
+  private uploadService = inject(UploadService);
+  rendererFactory = inject(RendererFactory2);
```

Comment:
Can you verify if moving the rendererFactory inject to inside the constructor still works? According to the previous implementation, the rendererFactory instance was only available inside the constructor, and would free up its occupied memory after constructor code execution. However, with the current implementation, it would keep occupying its memory, as long as the service/component/etc is alive, even though its only really being used inside the constructor